Networking/Networking/Remote/GenerativeContentRemote.swift

```diff
@@ -30,6 +30,30 @@ public protocol GenerativeContentRemoteProtocol {
     func identifyLanguage(siteID: Int64,
                           string: String,
                           feature: GenerativeContentRemoteFeature) async throws -> String
+
+    /// Generates a product using provided info
+    /// - Parameters:
+    ///   - siteID: WPCOM ID of the site.
+    ///   - productName: Product name to input to AI prompt
+    ///   - keywords: Keywords describing the product to input for AI prompt
+    ///   - language: Language to generate the product details
+    ///   - tone: Tone of AI - Represented by `AIToneVoice`
+    ///   - currencySymbol: Currency symbol to generate product price
+    ///   - dimensionUnit: Weight unit to generate product dimensions
+    ///   - weightUnit: Weight unit to generate product weight
+    ///   - categories: Existing categories
+    ///   - tags: Existing tags
+    /// - Returns: Generated `Product`
+    func generateProduct(siteID: Int64,
+                         productName: String,
+                         keywords: String,
+                         language: String,
+                         tone: String,
+                         currencySymbol: String,
+                         dimensionUnit: String,
+                         weightUnit: String,
+                         categories: [ProductCategory],
+                         tags: [ProductTag]) async throws -> Product
```

Comment:
❓ I wonder if we should create the prompt directly in the Yosemite layer and send it to the Networking layer afterward. With that, we would not have to update both layers when we need to change something to the prompt. WDYT?

Reply:
I plan to pass the `categories` and `tags` to the Networking layer and create the final `Product` inside the mapper class. i.e. Even if we create the prompt here, we will still have to send the `categories` and `tags` to the Networking layer. 


> With that, we would not have to update both layers when we need to change something to the prompt. WDYT?

- If we aren't adding any new parameters we can update the `Networking` layer when we need to make changes to the prompt. 
- We also can update the mapper when we want to make changes to how we handle the response to create a `Product`

What do you think?

Reply:
I understand. I will update the PR description. Thank you!